java/client/src/main/java/glide/api/models/commands/geospatial/GeoSearchOptions.java

```diff
@@ -0,0 +1,90 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.geospatial;
+
+import glide.api.commands.GeospatialIndicesBaseCommands;
+import glide.api.models.commands.ConditionalChange;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Optional arguments for {@link GeospatialIndicesBaseCommands#geoadd(String, Map, GeoAddOptions)}
+ * command.
+ *
+ * @see <a href="https://redis.io/commands/geoadd/">redis.io</a>
+ */
+public final class GeoSearchOptions {
+    /** Redis API keyword used to perform geosearch with coordinates. */
+    public static final String WITHCOORD_REDIS_API = "WITHCOORD";
+
+    /** Redis API keyword used to perform geosearch with distance. */
+    public static final String WITHDIST_REDIS_API = "WITHDIST";
+
+    /** Redis API keyword used to perform geosearch with hash value. */
+    public static final String WITHHASH_REDIS_API = "WITHHASH";
+
+    /** Options for handling existing members. See {@link ConditionalChange}. */
+    private final boolean withCoord;
+
+    private final boolean withDist;
+
+    private final boolean withHash;
+
+    public GeoSearchOptions(builder builder) {
+        this.withDist = builder.withDist;
+        this.withCoord = builder.withCoord;
+        this.withHash = builder.withHash;
+    }
+
+    /**
+     * Converts GeoAddOptions into a String[].
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        List<String> arguments = new ArrayList<>();
+
+        if (withDist) {
+            arguments.add(WITHDIST_REDIS_API);
+        }
+
+        if (withCoord) {
+            arguments.add(WITHCOORD_REDIS_API);
+        }
+
+        if (withHash) {
+            arguments.add(WITHHASH_REDIS_API);
+        }
+
+        return arguments.toArray(new String[0]);
+    }
+
+    public static class builder {
```

Comment:
use `@Builder` and delete this

Reply:
I felt like using lomboks version would allow the user to do like builder.withcoord(false) which is essentially no op, I wanted to make it so that if this builder setter is called its to enable that option, dont think i can do that with `@Builder`